pkg/sql/sql.go

```diff
@@ -264,8 +264,9 @@ func (c *Client) DropUser(ctx context.Context, accountName string) error {
 	return c.ExecFlushingPrivileges(ctx, query)
 }
 
-func (c *Client) AlterUser(ctx context.Context, accountName, password string) error {
-	query := fmt.Sprintf("ALTER USER '%s' IDENTIFIED BY '%s';", accountName, password)
+func (c *Client) AlterUser(ctx context.Context, accountName, password string, maxConnections int32) error {
```

Comment:
I suggest the following signature:

```go
type UserOpts struct {
	IdentifiedBy       string
	MaxUserConnections int32
}

type UserOpt func(*UserOpts)

func WithIdentifiedBy(password string) UserOpt {
	return func(cuo *UserOpts) {
		cuo.IdentifiedBy = password
	}
}

func WithMaxUserConnections(maxConns int32) UserOpt {
	return func(cuo *UserOpts) {
		cuo.MaxUserConnections = maxConns
	}
}

func (c *Client) AlterUser(ctx context.Context, accountName string, userOpts ...UserOpt) error
```

Basically re-use the same options as [CreateUser](https://github.com/mariadb-operator/mariadb-operator/blob/4788a7e43380b6fb2000bbb64224e60ca8c070b6/pkg/sql/sql.go#L240) for consistency.

Reply:
Options should only by set if non empty, like we do in [CreateUser](https://github.com/mariadb-operator/mariadb-operator/blob/4788a7e43380b6fb2000bbb64224e60ca8c070b6/pkg/sql/sql.go#L240)

Reply:
Thanks for the insight! I've tried to re-use the same options as `CreateUserOpts{}`, it also takes into account to avoid altering the `MAX_USER_CONNECTIONS` whenever the `maxconnections` param is 0.
```
func (c *Client) AlterUser(ctx context.Context, accountName string, createUserOpts ...CreateUserOpt) error {
	opts := CreateUserOpts{}
	for _, setOpt := range createUserOpts {
		setOpt(&opts)
	}

	query := fmt.Sprintf("ALTER USER '%s' ", accountName)
	if opts.IdentifiedBy != "" {
		query += fmt.Sprintf("IDENTIFIED BY '%s' ", opts.IdentifiedBy)
	}
	if opts.MaxUserConnections > 0 {
		query += fmt.Sprintf("WITH MAX_USER_CONNECTIONS %d ", opts.MaxUserConnections)
	}
	query += ";"

	return c.ExecFlushingPrivileges(ctx, query)
}
```